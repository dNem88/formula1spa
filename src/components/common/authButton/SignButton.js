import React from 'react';
import {Link} from 'react-router-dom'
import styles from './signButton.module.css'
function SignButton({textContent, href}) {

    return (
        <Link className={styles.link} to={href}>{textContent}</Link>
    )
}

export default SignButton;