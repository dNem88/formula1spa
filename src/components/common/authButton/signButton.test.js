import React from 'react'
import renderer from 'react-test-renderer';
import SignButton from './SignButton'
import {MemoryRouter, Link} from 'react-router-dom'

describe('Sign up /Sign in in button test ', () => {
    
    it('Should redirect to Login Page', () => {
        let linkComp = renderer.create(
            <MemoryRouter>
                <SignButton textContent={'Sign in'} href={'/login'}/>
            </MemoryRouter>
        ).toJSON();
        expect(linkComp).toMatchSnapshot();
        expect(linkComp.props.href).toBe('/login');
        expect(linkComp.props.className).toBe('link');
    })
    it('Should redirect to Login Page', () => {
        let linkComp = renderer.create(
            <MemoryRouter>
                <SignButton textContent={'Sign up'} href={'/register'}/>
            </MemoryRouter>
        ).toJSON();
        expect(linkComp).toMatchSnapshot();
        expect(linkComp.props.href).toBe('/register');
        expect(linkComp.props.className).toBe('link');
    })
})