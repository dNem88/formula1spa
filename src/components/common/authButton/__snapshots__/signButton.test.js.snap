// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sign up /Sign in in button test  Should redirect to Login Page 1`] = `
<a
  className="link"
  href="/login"
  onClick={[Function]}
>
  Sign in
</a>
`;

exports[`Sign up /Sign in in button test  Should redirect to Login Page 2`] = `
<a
  className="link"
  href="/register"
  onClick={[Function]}
>
  Sign up
</a>
`;
